modules/bmc/ipmi.rb

```diff
@@ -5,22 +5,41 @@ module Proxy
   module BMC
     class IPMI < Base
       include Proxy::Log
+      attr_accessor :logger, :log_level
+
+      # set the log level of rubyipmi, set to 0 or Logger::DEBUG for usablable output
+      def self.log_level=(log_level)
+        Rubyipmi.log_level = log_level
+      end
+
+      def self.log_level
+        Rubyipmi.logger.level
+      end
+
+      def self.logger
+        Rubyipmi.logger
+      end
+
+      # set the logger for rubyipmi
+      def self.logger=(log)
+        Rubyipmi.logger = log
+      end
 
       def self.installed?(provider)
         # check with the lib to see if at least one provider is installed
-        if provider
+        begin
           Rubyipmi.is_provider_installed?(provider)
-        else
-          Rubyipmi.provider_installed?
+        rescue
+          false
         end
       end
 
-      def self.providers_installed?
-        return Rubyipmi.providers_installed?
+      def self.providers_installed
```

Comment:
Yes, I know.  I noticed this the other day in Rubyipmi and I plan to fix it but it will have to wait since its a breaking change.  I have updated this in smart-proxy but this this will most likely be fixed in 0.11.0 of rubyipmi.